src/driver/linux_resource/syscall_aarch64.c

```diff
@@ -57,70 +233,211 @@ static typeof(aarch64_insn_adrp_get_offset) *aarch64_insn_adrp_get_offset_sym;
  * be sure it will never break, so there are as many safety checks here as possible.
  */
 
+/* Modern kernels way (since 5.4) (tested on 5.15.83):
+ * - Instead of `el_sync` we look for `el0t_64_sync`. Some new code was added
+ *   to the exception handler. See find_el_sync() for more comments.
+ * - `el0t_64_sync` jumps to C function el0t_64_sync_handler(), which calls
+ *   specific function based on the value of esr_el1 system register. We look
+ *   for `el0_svc` here. See find_el_svc_entry() for more comments.
+ * - `el0_svc()` calls `do_el0_svc()`, which calls `el0_svc_common()` and passes
+ *   pointer to the syscall table as an argument. See find_syscall_table_via_vbar()
+ *   for more comments.
+ */
+
 static ci_uint8 *
 find_el_sync(void)
 {
   ci_uint8 *vectors;
   ci_uint32 insn;
-  ci_uint8 *el_sync = efrm_find_ksym("el0_sync");
+  ci_uint8 *el_sync = NULL;
 
+#ifdef EFRM_HAVE_NEW_KALLSYMS
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 13, 0)
```

Comment:
Hmm, I thought `EFRM_HAVE_NEW_KALLSYMS` was removed in 5.7.

Reply:
Thank you for pointing it out. Indeed, this extra code is not needed.